src/gui/plugins/component_inspector/Light.qml

```diff
@@ -0,0 +1,1083 @@
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: 0
+      maximumValue: 1
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+  Component {
+    id: spinZeroMax
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: 0
+      maximumValue: 1000000
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+  Component {
+    id: spinNoLimit
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: -100000
+      maximumValue: 100000
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+
+  Component {
+    id: readOnlyNumber
+    Text {
+      id: numberText
+      anchors.fill: parent
+      horizontalAlignment: Text.AlignRight
+      verticalAlignment: Text.AlignVCenter
+      text: {
+        var decimals = numberText.width < 100 ? 2 : 6
+        return numberValue.toFixed(decimals)
+      }
+    }
+  }
+
+  Column {
+    anchors.fill: parent
+
+    // Header
+    Rectangle {
+      id: header
+      width: parent.width
+      height: typeHeader.height
+      color: "transparent"
+
+      RowLayout {
+        anchors.fill: parent
+        Item {
+          width: margin
+        }
+        Image {
+          id: icon
+          sourceSize.height: indentation
+          sourceSize.width: indentation
+          fillMode: Image.Pad
+          Layout.alignment : Qt.AlignVCenter
+          source: content.show ?
+              "qrc:/Gazebo/images/minus.png" : "qrc:/Gazebo/images/plus.png"
+        }
+        TypeHeader {
+          id: typeHeader
+        }
+        Item {
+          Layout.fillWidth: true
+        }
+      }
+      MouseArea {
+        anchors.fill: parent
+        hoverEnabled: true
+        cursorShape: Qt.PointingHandCursor
+        onClicked: {
+          content.show = !content.show
+        }
+        onEntered: {
+          header.color = highlightColor
+        }
+        onExited: {
+          header.color = "transparent"
+        }
+      }
+    }
+
+    // Content
+    Rectangle {
+      id: content
+      property bool show: false
+      width: parent.width
+      height: show ? grid.height : 0
+      clip: true
+      color: "transparent"
+
+      Behavior on height {
+        NumberAnimation {
+          duration: 200;
+          easing.type: Easing.InOutQuad
+        }
+      }
+
+
+      GridLayout {
+        id: grid
+        width: parent.width
+        columns: 6
+
+        Text {
+          Layout.columnSpan: 6
+          text: "      Specular"
+          color: "dimgrey"
+          width: margin + indentation
+        }
+
+        // Left spacer
+        Item {
+          Layout.rowSpan: 3
+          width: margin + indentation
+        }
+
+        Component {
```

Comment:
How about moving this component up top with the other ones? Having it inside the layout makes the layout contents a bit messier to understand.